### PR TITLE
chore(backend): refactor `verifyWebhook()` to use `standardwebhooks` library

### PR DESCRIPTION
**Pull request by brkalow**
## Description

Aligning on the `standardwebhooks` library for webhook verification.

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/backend/src/webhooks.ts [hunk 1]
```diff
@@ -56,39 +86,42 @@ export * from './api/resources/Webhooks';
...
-  if (svixSignature.split(' ').includes(constructedSignature)) {
-    return errorThrower.throw('Incoming webhook does not have a valid signature');
+    return {
+      type: payload.type,
+      object: 'event',
+      data: payload.data,
+    } as WebhookEvent;
+  } catch (e) {
+    return errorThrower.throw(`Unable to verify incoming webhook: ${e instanceof Error ? e.message : 'Unknown error'}`);
   }
```
[the remark this answers is not in the document]
Comment (brkalow) [reply packages/backend/src/webhooks.ts:135]: Given that this payload comes from our webhook provider and we've verified that the signature is accurate, we have an implicit guarantee of the shape here.

**Comment by bratsos** [comment]
❓ According to [standardwebhooks](https://github.com/standard-webhooks/standard-webhooks/blob/main/spec/standard-webhooks.md#verifying-signatures) (TIL), it's a good practice to verify the webhook timestamp, making sure it's within a reasonable timeframe to prevent replay attacks. Do you think we should implement this here as well, or add this in a follow-up PR?

File: packages/backend/src/__tests__/webhooks.test.ts [hunk 2]
```diff
@@ -1,39 +1,45 @@
...
       headers: new Headers({
-        // Missing svix-signature but with valid format for others
-        'svix-id': 'msg_123',
-        'svix-timestamp': '1614265330',
+        // Missing all required headers
       }),
     });
 
-    await expect(verifyWebhook(mockRequest)).rejects.toThrow('Missing required Svix headers: svix-signature');
+    await expect(verifyWebhook(mockRequest)).rejects.toThrow('Missing required webhook headers');
```
Comment (bratsos) [inline packages/backend/src/__tests__/webhooks.test.ts:30]: 🙃 Maybe we can add a test for whitespace header values to handle this:

```js
headers: new Headers({
    'svix-id': '',        // Empty - would be caught
    'svix-timestamp': '   ', // Whitespace - would NOT be caught
    'svix-signature': 'v1,signature',
})
```

**Review by jacekradko — APPROVED** [review]

**Comment by brkalow** [comment]
> According to [standardwebhooks](https://github.com/standard-webhooks/standard-webhooks/blob/main/spec/standard-webhooks.md#verifying-signatures) (TIL), it's a good practice to verify the webhook timestamp, making sure it's within a reasonable timeframe to prevent replay attacks. Do you think we should implement this here as well, or add this in a follow-up PR?

Good call! The `standardwebhooks` package takes care of this internally already 👍

File: packages/backend/src/__tests__/webhooks.test.ts (hunk 2 above)
Comment (brkalow) [reply packages/backend/src/__tests__/webhooks.test.ts:30]: Good idea, updated!
